crates/bin/cli-wallet/src/main.rs

```diff
@@ -241,7 +246,26 @@ async fn main() -> Result<()> {
             let wad = opt_proofs
                 .map(|proofs| Wad { node_url, proofs })
                 .ok_or(anyhow!("Not enough funds"))?;
-            println!("Wad:\n{}", serde_json::to_string(&wad)?);
+
+            match output {
+                Some(output_path) => {
+                    if let Some(ext) = output_path.extension() {
+                        if ext != "json" {
+                            return Err(anyhow!("Error: Output file must have a .json extension."));
+                        }
+                    } else {
+                        return Err(anyhow!("Output file should be a `.json` file"));
+                    }
+
+                    fs::write(&output_path, serde_json::to_string_pretty(&wad)?).map_err(|e| {
+                        anyhow!("could not write tot file {:?}: {}", output_path, e)
+                    })?;
```

Comment:
Still, this can fail. So move the `tx.commit()?;` line at the very end, after we printed or written the tokens.
If it fails and we return an error, the tx won't be committed and we will be fine